packages/gatsby/src/utils/webpack/plugins/static-query-mapper.ts

```diff
@@ -14,6 +14,15 @@ import type {
 type ChunkGroup = Compilation["chunkGroups"][0]
 type EntryPoint = Compilation["asyncEntrypoints"][0]
 
+const removeExportQueryParam = (
+  path: string | undefined
+): string | undefined => {
+  if (path?.endsWith(`?export=head`) || path?.endsWith(`?export=default`)) {
```

Comment:
Are we sure this is robust enough? I know `resourcePath` is compared against this below, but it seems like we either need this to outright remove all query params (and update the name of the function for future us) or support just removing what's exactly needed.